src/ralph/models/converter.py

```diff
@@ -0,0 +1,82 @@
+"""Base converter definition"""
+
+from abc import ABC, abstractmethod
+from uuid import UUID, uuid5
+
+from pydantic import BaseModel
+
+from ralph.defaults import MODEL_PATH_SEPARATOR
+from ralph.exceptions import ConversionException
+from ralph.utils import get_dict_value_from_path, set_dict_value_from_path
+
+
+class BaseConverter(ABC):
+    """Converter Base Class.
+
+    Converters define a conversion set to convert from one event type to another.
+    """
+
+    __model__: BaseModel
+
+    def __init__(self, uuid_namespace: UUID):
+        """Initializes BaseConverter."""
+
+        self.uuid_namespace = uuid_namespace
+        self.field_array = []
+        self.prepare_converter(self.field_array, self.get_conversion_set())
+
+    @abstractmethod
+    def get_conversion_set(self):
+        """Returns a conversion set used for conversion.
+
+        Each element in the conversion set should be a tuple of length 2 or 3.
+        - The first element of each tuple is a destination path string.
+        - The second element can be a source path string or a function with no arguments.
+        - The optional third element defines a transformer function with one argument.
+        """
+
+    def convert(self, event: dict, event_str: str):
+        """Converts and serializes the event.
+
+        Args:
+            event (dict): the event to convert and serialize.
+            event_str (str): the event original string value.
+        Returns:
+            (model): converted pydantic model.
+        """
+
+        converted_event = {}
+        for destination_path, source_path, transformer in self.field_array:
+            field = get_dict_value_from_path(event, source_path)
```

Comment:
This was confusing, sorry, the source_path switched with transformer in case it were callable in the prepare method. Now, with constant tuple length of 3 this is not  the case anymore)